code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,658 @@
+          $ref: '#/components/responses/500'
+        '503':
+          $ref: '#/components/responses/503'
+ 
+  /edgeCloudPlatform:
+    get:
+      tags: 
+        - Mobile Edge Cloud Platform
+      summary: Retrieve a list of the operator’s MECs and their status
+      description: List of the operator’s MECs and their status, ordering the results by location and filtering by status (active/inactive/unknown)
+      operationId: getMecPlatforms
+      parameters:
+        - name: regionId
+          description: Human readable name of the geografical zone of the MEC. Defined by the OP.
+          in: query
+          required: false
+          schema: 
+              $ref: '#/components/schemas/RegionId'
+        - name: status
+          in: query
+          required: false
+          schema: 
+              $ref: '#/components/schemas/Status'
+      responses:
+        '200':
+          description: Available MEC Platforms 
+          content:
+            application/json:
+              schema:
+                type: array
+                items:                  
+                  $ref: '#/components/schemas/MecDetails'
+                minItems: 1
+        '401':
+          $ref: '#/components/responses/401'
+        '403':
+          $ref: '#/components/responses/403'
+        '404':
+          $ref: '#/components/responses/404'
+        '500':
+          $ref: '#/components/responses/500'
+        '503':
+          $ref: '#/components/responses/503'
+ 
+components:
+  securitySchemes:
+    oAuth2ClientCredentials:
+      type: oauth2
+      flows: 
+        clientCredentials: 
+          tokenUrl: '/oauth2/token'
+          scopes:
+            nbi-mgmt: Access to the APIs
+  schemas:
+    AppId:
+      type: string
+      format: uuid
+      description: A globally unique identifier associated with the artifact. Originating OP generates this identifier when artifact is submitted over NBI.
+    AppInstanceId:
+      type: string
+      format: uuid
+      description: A globally unique identifier associated with a running instance of an application. Originating OP generates this identifier. 
+    AppProviderId:
+      type: string
+      pattern: ^[A-Za-z][A-Za-z0-9_]{7,63}$
+      description: UserId of the app provider.  Identifier is relevant only in context of federation.
+    AppProvisioning:
+      description: Attributes required to provision an application
+      type: object
+      properties:
+        artifactId:
+          $ref: '#/components/schemas/ArtifactId'
+        appName:
+          $ref: '#/components/schemas/AppName'
+        edgeResourceName:
+          $ref: '#/components/schemas/EdgeResourceName'
+        regionId:
+          $ref: '#/components/schemas/RegionId'
+      required:
+        - artifactId
+        - appName
+    AppName:
+      type: string
+      description: Name given by the application provider
+    ArtifactId:
+      type: string
+      pattern: 6sa85a64-5117-3562-c3fd-2c913a66aba1
```

Comment:
pattern should be a regular expression, or change it to "format: uuid".

Reply:
AppProviderId is no longer being used as discussed above. Anyhow ArtifactId has been modified